tests/python_tests/test_llm_pipeline.py

```diff
@@ -842,3 +842,50 @@ def test_pipelines_with_gguf_generate(pipeline_type, model_ids):
     res_string_input_2 = hf_tokenizer.batch_decode([encoded_result.tokens[0]], skip_special_tokens=True)[0]
 
     assert res_string_input_1 == res_string_input_2
+
+@pytest.mark.parametrize("pipeline_type", get_gguf_pipeline_types())
+@pytest.mark.parametrize("model_ids", get_gguf_model_list())
+@pytest.mark.parametrize("enable_save_ov_model", get_gguf_enable_save_ov_model_property_list())
+@pytest.mark.precommit
+
+def test_pipelines_with_gguf_generate_with_enable_save_ov_model_property(pipeline_type, model_ids, enable_save_ov_model):
+    gguf_model_id = model_ids["gguf_model_id"]
+    gguf_filename = model_ids["gguf_filename"]
+    prompt = 'Why is the Sun yellow?'
+
+    from utils.network import retry_request
+    from transformers import AutoTokenizer, AutoModelForCausalLM
+    from openvino_tokenizers import convert_tokenizer
+
+    opt_model = retry_request(lambda: AutoModelForCausalLM.from_pretrained(gguf_model_id, gguf_file=gguf_filename))
+    hf_tokenizer = retry_request(lambda: AutoTokenizer.from_pretrained(gguf_model_id, gguf_file=gguf_filename))
+
+    ov_generation_config = ov_genai.GenerationConfig()
+    ov_generation_config.max_new_tokens = 30
+    ov_generation_config.apply_chat_template = False
+    ov_generation_config.set_eos_token_id(hf_tokenizer.eos_token_id)
+
+    inputs = hf_tokenizer(prompt, return_tensors="pt")
+    input_ids, attention_mask = inputs['input_ids'], inputs['attention_mask']
+    hf_generation_config = generation_config_to_hf(opt_model.generation_config, ov_generation_config)
+    generate_outputs = opt_model.generate(input_ids=input_ids, attention_mask=attention_mask, generation_config=hf_generation_config, tokenizer=hf_tokenizer)
+    prompt_len = 0 if ov_generation_config.echo else input_ids.numel()
+    all_text_batch = hf_tokenizer.batch_decode([generated_ids[prompt_len:] for generated_ids in generate_outputs.sequences], skip_special_tokens=True)
+    res_string_input_1 = all_text_batch[0]
+
+    ov_tokenizer, ov_detokenizer = convert_tokenizer(hf_tokenizer, with_detokenizer=True)
+
+    gguf_full_path = download_gguf_model(gguf_model_id, gguf_filename)
+    gguf_full_path = Path(gguf_full_path)
+    ov_tokenizer_path = gguf_full_path.parent / "openvino_tokenizer.xml"
+    ov_detokenizer_path = gguf_full_path.parent / "openvino_detokenizer.xml"
+    ov.save_model(ov_tokenizer, ov_tokenizer_path)
+    ov.save_model(ov_detokenizer, ov_detokenizer_path)
+    ov_pipe_gguf = create_ov_gguf_pipeline(gguf_full_path, pipeline_type=pipeline_type, enable_save_ov_model=enable_save_ov_model)
+    encoded_result  = ov_pipe_gguf.generate(ov.Tensor(input_ids.numpy()), generation_config=ov_generation_config)
+    res_string_input_2 = hf_tokenizer.batch_decode([encoded_result.tokens[0]], skip_special_tokens=True)[0]
+
+    ov_model_saved_on_disk = False
+    if Path(gguf_full_path / "openvino_model.xml").exists():
+        ov_model_saved_on_disk = True
+    assert enable_save_ov_model == ov_model_saved_on_disk
```

Comment:
I would propose to test the following scenario:
Construct GGUF pipeline. Construct another pipeline after that from `.xml`

That's the user scenario you are trying to cover with this PR, right?

Reply:
Update test case to construct two pipeline 
1.  Construct GGUF pipeline and save generated ov model
2.  Construct OV pipeline from generated ov model
3.  Compare pipeline 1 & pipeline 2 generated results.